components/properties/src/props.rs

```diff
@@ -41,6 +49,147 @@ enum EnumeratedProperty {
     InvalidCode = -1, // TODO(#1160) - taken from ICU4C UProperty::UCHAR_INVALID_CODE
 }
 
+/// Private marker type for PropertyValueNameToEnumMapper
+/// to work for all properties at once
+#[derive(Clone, Copy, PartialEq, Eq, Hash, Debug)]
+pub(crate) struct ErasedValueNameMap;
```

Comment:
Nit:
```suggestion
pub(crate) struct ErasedPropertyValueNameToEnumMapV1Marker;
```

Reply:
Made it a bit shorter; it's a private type and the name doesn't matter as much